### PR TITLE
Add rectified flow noise scheduler for accelerated diffusion model

### PR DESCRIPTION
**Pull request by Can-Zhao**
Fixes # .

### Description

A few sentences describing the changes proposed in this pull request.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by KumoLiu — APPROVED** [review]
Thanks for the PR, overall looks good to me.
One minor comment and please also resolve the conflict, then I can help trigger the blossom. Thanks.

File: monai/inferers/inferer.py [hunk 1]
```diff
@@ -859,12 +859,19 @@ def sample(
...
+
+        all_next_timesteps = torch.cat((scheduler.timesteps[1:], torch.tensor([0], dtype=scheduler.timesteps.dtype)))
         if verbose and has_tqdm:
-            progress_bar = tqdm(scheduler.timesteps)
+            progress_bar = tqdm(
+                zip(scheduler.timesteps, all_next_timesteps),
+                total=min(len(scheduler.timesteps), len(all_next_timesteps)),
+            )
         else:
             progress_bar = iter(scheduler.timesteps)
```
[the remark this answers is not in the document]
Comment (KumoLiu) [reply monai/inferers/inferer.py:870]: ? This line looks redundant, if you add Line871

[the remark this answers is not in the document]
Comment (Can-Zhao) [reply monai/inferers/inferer.py]: Ah, nice catch. Thank you!

File: monai/networks/schedulers/rectified_flow.py [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+            noise: noise to add to samples
+            timesteps: timesteps tensor indicating the timestep to be computed for each sample.
+
+        Returns:
+            noisy_samples: sample with added noise
+        """
+        timepoints: torch.Tensor = timesteps.float() / self.num_train_timesteps
+        timepoints = 1 - timepoints  # [1,1/1000]
+
+        # timepoint  (bsz) noise: (bsz, 4, frame, w ,h)
```
[the remark this answers is not in the document]
Comment (Can-Zhao) [reply monai/networks/schedulers/rectified_flow.py]: Nice catch. Thanks!

File: monai/inferers/inferer.py [hunk 3]
```diff
@@ -39,7 +39,7 @@
     SPADEAutoencoderKL,
     SPADEDiffusionModelUNet,
```
Comment (virginiafdez) [inline monai/inferers/inferer.py:40]: Even if this scheduler is only used for sampling (same as PNDM), is it incompatible with latent diffusion, controlnet diffusion and controlnet latent diffusion inferers? If not, these inferers' sample method can also be tested with this scheduler. This means the ControlNetDiffusionInferer has to be modified. I think the latent methods should be fine, as they call the normal one, but adding tests on test_latent_diffusion_inferer.py and test_controlnet_inferers.py would be a good way to ensure these schedulers can be used with them as well.

**Review by virginiafdez — COMMENTED** [review]
I am not very familiar with this type of method. From the code that has been modified, I assume this scheduler is not meant to be used to train diffusion models, but merely sampling from them (such as PNDM)? 
Even in this case, they could be used with latent diffusion models and controlnets, right? Could these inferers be tested with this scheduler as well?

File: monai/networks/schedulers/rectified_flow.py [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+        # expand timepoint to noise shape
+        timepoints = timepoints.unsqueeze(1).unsqueeze(1).unsqueeze(1).unsqueeze(1)
+        timepoints = timepoints.repeat(1, noise.shape[1], noise.shape[2], noise.shape[3], noise.shape[4])
```
[the remark this answers is not in the document]
Comment (Can-Zhao) [reply monai/networks/schedulers/rectified_flow.py]: Yes! Thank you so much!

File: monai/inferers/inferer.py (hunk 3 above)
Comment (Can-Zhao) [reply monai/inferers/inferer.py:40]: That's very valuable! I just added the tests. Thank you so much!

File: tests/inferers/test_diffusion_inferer.py [hunk 5]
```diff
@@ -19,7 +19,7 @@
 
 from monai.inferers import DiffusionInferer
```
[the remark this answers is not in the document]
Comment (Can-Zhao) [reply tests/inferers/test_diffusion_inferer.py:20]: That's true. It can be used for training. I added test. Thank you a lot!

**Review by virginiafdez — COMMENTED** [review]
Looks fine by me!
